frontend/src/tests/wage-chunk-tests.js

```diff
@@ -1,12 +1,221 @@
-import { it } from "./utils";
+import PropTypes from "prop-types";
+import {
+    checkPropTypes,
+    wageChunkPropTypes,
+    expect,
+    it,
+    beforeAll,
+} from "./utils";
+
+import { databaseSeeder } from "./setup";
 
 // TODO: Remove eslint disable
 // eslint-disable-next-line
-export function wageChunksTests({ apiGET, apiPOST }) {
-    it.todo("get wage_chunks for position");
-    it.todo("create and delete wage_chunk for position");
-    it.todo("update wage_chunk");
-    it.todo(
-        "automatic creation of wage chunk for position when `hours` is set"
-    );
+export function wageChunksTests(api) {
+    const { apiGET, apiPOST } = api;
+    const session = databaseSeeder.seededData.session,
+        applicant = databaseSeeder.seededData.applicant,
+        contractTemplate = databaseSeeder.seededData.contractTemplate,
+        assignment = databaseSeeder.seededData.assignment;
+
+    let originalWageChunk;
+
+    beforeAll(async () => {
+        await databaseSeeder.seed(api);
+    });
+
+    it("get wage_chunks for position", async () => {
+        const resp = await apiGET(
+            `/admin/assignments/${assignment.id}/wage_chunks`
+        );
+        expect(resp).toHaveStatus("success");
+
+        const wageChunkData = {
+            hours: assignment.hours,
+        };
+
+        // make sure the position we created is in that list
+        expect(resp.payload).toContainObject(wageChunkData);
+
+        checkPropTypes(PropTypes.arrayOf(wageChunkPropTypes), resp.payload);
+
+        originalWageChunk = resp.payload[0];
```

Comment:
What if the test fails before this line is executed? Will that not screw up future tests that depend on `originalWageChunk` being a certain value?

Reply:
`originalWageChunk` is mostly working as a template in the test. We use `originalWageChunk` as a template to create new wage chunk object so that if any required field is added to wage chunk, the value from `originalWageChunk` apply to the new wage chunk and it won't affect the test. 

It is also used to check if `originalWageChunk` is still in the assignment wage chunk list.

The place where it can screw up future tests would be if `get` method does not return anything. In this case, it would be fine for all the wage chunk test to fail since every wage chunk test case calls the `get` method at the end.